src/ngraph/pass/manager.hpp

```diff
@@ -44,6 +44,11 @@ class ngraph::pass::Manager
     template <typename T, class... Args>
     std::shared_ptr<T> register_pass(Args&&... args)
     {
+        auto pass = get_pass_name(typeid(T).name());
```

Comment:
since type_info.name() produces an undefined output it can't be used to get the pass name. You need some other way to do this.

Reply:
`Why put all of this code in the pass manager when all you need to do is at the point where you populate the pass::Manager just don't register passes`

@rkimballn1 this check will have to be done by each backend. Currently CPU backend does this. This PR just moves/generalizes that logic into core, so all backends can take advantage of this env var filter for passes. `typeid(T).name()` is previously used in code in this file, just reusing here.

Reply:
No, typeid was never used for `functional` code where the value returned is used in a conditional. You can't do that. It is perfectly valid for typeid(T).name() to return a null string.
I am very against this PR. The code should be done in the backend. Don't generalize when it makes things too complex.

Reply:
@rkimballn1 not a crucial feature, just a utility to make it easier if user wants to bypass some passes (even though backend enabled them).

Backends can explicitly enable this as needed (as CPU does), closing this PR